core-bundle/tests/Image/IntegrationTest.php

```diff
@@ -0,0 +1,1688 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\CoreBundle\Tests\Image;
+
+use Contao\Config;
+use Contao\Controller;
+use Contao\CoreBundle\Framework\Adapter;
+use Contao\CoreBundle\Image\ImageFactory;
+use Contao\CoreBundle\Image\LegacyResizer;
+use Contao\CoreBundle\Image\PictureFactory;
+use Contao\CoreBundle\Image\Studio\Studio;
+use Contao\CoreBundle\Tests\TestCase;
+use Contao\FilesModel;
+use Contao\FrontendTemplate;
+use Contao\Image\PictureGenerator;
+use Contao\Image\ResizeCalculator;
+use Contao\ImagineSvg\Imagine as ImagineSvg;
+use Contao\LayoutModel;
+use Contao\Model\Registry;
+use Contao\PageModel;
+use Contao\System;
+use Contao\Template;
+use Imagine\Gd\Imagine as ImagineGd;
+use Psr\Log\NullLogger;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+use Symfony\Component\DependencyInjection\ParameterBag\ParameterBag;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpFoundation\RequestStack;
+use Webmozart\PathUtil\Path;
+
+class IntegrationTest extends TestCase
+{
+    private static $testRoot;
+
+    public static function setUpBeforeClass(): void
+    {
+        parent::setUpBeforeClass();
+
+        // Make resources available in test root
+        self::$testRoot = self::getTempDir();
+
+        $filesystem = new Filesystem();
+
+        $filesystem->symlink(
+            Path::canonicalize(__DIR__.'/../Fixtures/files'),
+            Path::join(self::$testRoot, 'files')
+        );
+
+        $filesystem->symlink(
+            Path::canonicalize(__DIR__.'/../../src/Resources/contao'),
+            Path::join(self::$testRoot, 'contao')
+        );
+    }
+
+    /**
+     * @dataProvider provideControllerAddImageToTemplateTestCases
+     *
+     * @group legacy
+     */
+    public function testControllerAddImageToTemplate(\Closure $testCase, array $expectedTemplateData): void
+    {
+        [$template, $dataRow, $maxWidth, $lightBoxGroupIdentifier, $filesModel] = $this->setUpTestCase($testCase);
+
+        Controller::addImageToTemplate($template, $dataRow, $maxWidth, $lightBoxGroupIdentifier, $filesModel);
+
+        $this->assertSameTemplateData($expectedTemplateData, $template);
+    }
+
+    // fixme: Uncomment the following method to test + compare against old implementation / also see #1862.
```

Comment:
This needs to be addressed.

Reply:
This should be removed together with the `Controller::addImageToTemplate__old` before merging.

see https://github.com/contao/contao/pull/1753#discussion_r455138046

Reply:
Finally dropped in 1b670e533d5c03d21673c0d7c32c0154714a8a15